cnf-tests/testsuites/e2esuite/dpdk/dpdk.go

```diff
@@ -173,6 +178,107 @@ var _ = Describe("dpdk", func() {
+					}
+				}
+			}
+		})
+
+		BeforeEach(func() {
+			if discovery.Enabled() && !policyHasVhostnet {
+				Skip("Missing SriovNetworkNodePolicy with NeedVhostNet enabled")
+			}
+		})
+		Context("Client should be able to forward packets", func() {
+			It("Should be able to transmit packets", func() {
+				var out string
+				var err error
+
+				command := CREATE_TAP_DEVICE_COMMAND + `
+				dpdk-testpmd --vdev net_tap0,iface=tap23 --no-pci -- -ia --forward-mode txonly
+				sleep INF
+				`
+				txDpdkWorkloadPod, err := createDPDKWorkload(nodeSelector,
+					command,
+					false,
+					[]corev1.Capability{"NET_ADMIN"},
+					dpdkWorkloadMac,
+				)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Parsing output from the client DPDK application")
+				Eventually(func() string {
+					out, err = pods.GetLog(txDpdkWorkloadPod)
+					Expect(err).ToNot(HaveOccurred())
+					return out
+				}, 2*time.Minute, 1*time.Second).Should(ContainSubstring(LOG_ENTRY),
+					"Cannot find accumulated statistics")
+				By("Checking the tx output from the client DPDK application")
+				checkTxOnly(out)
+			})
+		})
+
+		Context("Server should be able to receive packets and forward to tap device", func() {
+			It("Should be able to transmit packets", func() {
+				var err error
+				var out string
+				// --stats-period is used to keep the command alive once the pod is started
+				serverCommand := fmt.Sprintf(`
+%s
+dpdk-testpmd --vdev net_tap0,iface=tap23 -w ${PCIDEVICE_OPENSHIFT_IO_%s} -- --stats-period 5
+sleep INF
+				`, CREATE_TAP_DEVICE_COMMAND, strings.ToUpper(dpdkResourceName))
+				dpdkWorkloadPod, err := createDPDKWorkload(nodeSelector, serverCommand, false, []corev1.Capability{"NET_ADMIN"}, dpdkWorkloadMac)
+				Expect(err).ToNot(HaveOccurred())
+
+				clientCommand := fmt.Sprintf(`
+dpdk-testpmd -w ${PCIDEVICE_OPENSHIFT_IO_%s} -- --forward-mode txonly --eth-peer=0,%s --stats-period 5
+sleep INF
+				`, strings.ToUpper(dpdkResourceName), dpdkWorkloadMac)
+				_, err = createDPDKWorkload(nodeSelector,
+					clientCommand,
+					false,
+					[]corev1.Capability{"NET_ADMIN"},
+					dpdkWorkloadMac,
+				)
+				Expect(err).ToNot(HaveOccurred())
+
+				Eventually(func() string {
+					out, err = pods.GetLog(dpdkWorkloadPod)
+					Expect(err).ToNot(HaveOccurred())
+					return out
+				}, 2*time.Minute, 1*time.Second).Should(ContainSubstring("Port statistics"),
+					"Cannot find port statistics")
+
+				By("Checking the rx output of tap device from the client DPDK application")
+				bytes, err := getDeviceRXBytes(dpdkWorkloadPod, "tap23")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(bytes).To(BeNumerically(">", 0))
+
+				By("Parsing output from the DPDK application")
+				Eventually(func() string {
+					out, err = pods.GetLog(dpdkWorkloadPod)
+					Expect(err).ToNot(HaveOccurred())
+					return out
+				}, 8*time.Minute, 1*time.Second).Should(ContainSubstring("NIC statistics for port"),
+					"Cannot find accumulated statistics")
+				checkRx(out)
```

Comment:
should the function be `checkRxOnly` like `checkTxOnly`

Reply:
done